### ISSUE TITLE
Tinker rearm bug

### ISSUE THREAD

**Issue by Imdonewithdota** [ValveSoftware/Dota2-Gameplay#11620]
### Ability name

Rearm

### Description

I was playing with my friend ranked game and everything was going fine but after 39 minutes my ability rearm( tinker) suddenly stopped functioning properly and ruined whole game. I tried all methods in the game to restore it but i failed in all attempts.

### Example Match ID (and possibly Timestamp)

7322508358 - 39:00

**Comment by akhil189** [comment]
This has been happening to me since last 2 days. I was spamming Tinker and I noticed that whenever we disconnect and reconnect this is happening. So far, this has happened to me thrice. 

e.g. I press key for rearm and I press key for blink immediately, the hero will blink once rearm channeling is completed. But, if I dc and reconnect during a game, pressing any key during rearm channeling is stopping the channeling and going to cooldown. I lost my games due to this.

@Imdonewithdota I'm confident this has happened to you when you disconnected and reconnected during the game. Do you remember?

**Comment by Imdonewithdota** [comment]
Yep you are absolutely right and i find a rather extreme solution which is just reinstall steam and dota because reinstalling dota is not enough.
